packages/expect/src/matchers.ts

```diff
@@ -887,17 +888,21 @@ const matchers: MatchersObject = {
     return {message, pass};
   },
 
-  toMatchObject(received: object, expected: object) {
+  toMatchObject(
+    received: object,
+    expected: object,
+    options?: MatchObjectOptions,
```

Comment:
I think for now there is no need to involve one more package just to push this type around. The `MatcherHintOptions` is different case, because `matcherHint()` logic lives in `'jest-matcher-utils'` package.

```suggestion
    options?: {ignoreElementsOrder?: boolean},
```